### PR TITLE
Serialization fixes, support attention mask in TransformerEncoder

### PR DESCRIPTION
**Pull request by mryab**

### CODE REVIEW

**Review by justheuristic — COMMENTED** [review]
Perhaps we should also perform .cpu() in hivemind.client.expert._RemoteModuleCall, whaddayasay?

**Review by justheuristic — APPROVED** [review]

File: hivemind/client/moe.py [hunk 1]
```diff
@@ -182,19 +184,20 @@ def forward(cls, ctx, dummy, experts_per_sample: List[List[RemoteExpert]], k_min
...
-        # list of torch tensors, where i-th tensor is of shape [num_responded, *expert_outputs[i].shape]
+        alive_flat_outputs_stacked = (torch.cat(outputs).to(flat_inputs[0].device)
+                                      for outputs in zip(*alive_flat_outputs))
+        # torch tensors, i-th tensor is of shape [num_responded, *expert_outputs[i].shape]
 
         outputs = []
         for response_stacked in alive_flat_outputs_stacked:
             output = torch.zeros(
-                [num_samples, max_experts, *response_stacked.shape[1:]], device=response_stacked.device,
+                [num_samples, max_experts, *response_stacked.shape[1:]], device=flat_inputs[0].device,
```
Comment (justheuristic) [inline hivemind/client/moe.py:194]: [not critical, not sure] - it may be faster to perform __setitem__ on cpu and then send results to GPU
